toucan_connectors/aircall/aircall_connector.py

```diff
@@ -31,42 +86,75 @@ class AircallConnector(ToucanConnector):
     bearer_integration = 'aircall_oauth'
     bearer_auth_id: str
 
-    def _get_page_data(
-        self, endpoint, query, jq_filter: str, page_number: int, per_page: int
-    ) -> Tuple[List[dict], bool]:
-        """Get the data for a single page and the information if the page is the last one"""
-        page_raw_data = self.bearer_oauth_get_endpoint(
-            endpoint, {**query, 'per_page': per_page, 'page': page_number}
-        )
-        try:
-            is_last_page = page_raw_data['meta']['next_page_link'] is None
-        except KeyError:
-            is_last_page = True
-        page_data = pyjq.first(jq_filter, page_raw_data)
-        if isinstance(page_data, dict):
-            page_data = [page_data]
-        return page_data, is_last_page
+    async def _get_data(self, dataset: str, limit) -> Tuple[List[dict], List[dict]]:
+        """Triggers fetches for data and does preliminary filtering process"""
+        headers = {'Authorization': BEARER_API_KEY, 'Bearer-Auth-Id': self.bearer_auth_id}
+        async with ClientSession(headers=headers) as session:
+            team_data, variable_data = await asyncio.gather(
+                fetch_page('teams', [], session, limit, 0,),
+                fetch_page(dataset, [], session, limit, 0,),
+            )
+
+            team_response_list = []
+            variable_response_list = []
+            if len(team_data) > 0:
+                for data in team_data:
+                    for team_obj in data.get('teams'):
+                        team_response_list += DICTIONARY_OF_FORMATTERS.get('teams')(team_obj)
```

Comment:
More or less same pattern with `.get()` here. Use square brackets directly. If `teams` is not in `data` or not in `DICTIONARY_OF_FORMATTERS`, your code crash with non obvious errors such as `NoneType is not iterable` or `Nonetype is not callabe`